service/history/timerQueueProcessor.go

```diff
@@ -800,10 +816,13 @@ Update_History_Loop:
 	return ErrMaxAttemptsExceeded
 }
 
-func (t *timerQueueProcessorImpl) updateWorkflowExecution(context *workflowExecutionContext,
-	msBuilder *mutableStateBuilder, scheduleNewDecision bool, timerTasks []persistence.Task,
-	clearTimerTask persistence.Task) error {
-	var transferTasks []persistence.Task
+func (t *timerQueueProcessorImpl) updateWorkflowExecution(
```

Comment:
The retries are done inside each process timer task like processWorkflowTimeout. The only thing there is retries are used for other logic as well(timer task specific work), unless it could be split in two retries once in process timer task and the other in update workflow, which complicates the logic.